source/extensions/transport_sockets/tls/context_config_impl.cc

```diff
@@ -179,7 +179,15 @@ ContextConfigImpl::ContextConfigImpl(
                                                 default_min_protocol_version)),
       max_protocol_version_(tlsVersionFromProto(config.tls_params().tls_maximum_protocol_version(),
                                                 default_max_protocol_version)),
-      handshaker_factory_(*HandshakerFactoryImpl::getDefaultHandshakerFactory()) {
+      handshaker_factory_cb_([&]() {
+        auto* factory = HandshakerFactoryImpl::getDefaultHandshakerFactory();
```

Comment:
I think this should go in an `else` after `if (config.has_custom_listener_handshaker()) {`.

Reply:
Done.